dask/array/overlap.py

```diff
@@ -584,29 +608,83 @@ def map_overlap(x, func, depth, boundary=None, trim=True, **kwargs):
            [20,  21,  22,  23],
            [24,  25,  26,  27]])
     """
-    depth2 = coerce_depth(x.ndim, depth)
-    boundary2 = coerce_boundary(x.ndim, boundary)
-
-    for i in range(x.ndim):
-        if isinstance(depth2[i], tuple) and boundary2[i] != "none":
-            raise NotImplementedError(
-                "Asymmetric overlap is currently only implemented "
-                "for boundary='none', however boundary for dimension "
-                "{} is {}".format(i, boundary2[i])
+    # Look for invocation using deprecated single-array signature
+    # map_overlap(x, func, depth, boundary=None, trim=True, **kwargs)
+    if isinstance(func, Array) and callable(args[0]):
+        warnings.warn(
+            "Detected use of signature map_overlap(x, func) rather than "
+            "map_overlap(func, *args) for multi-array support. Arguments "
+            "will be swapped in this case but such an exception will not "
+            "be made in a future release.",
+            FutureWarning,
+        )
```

Comment:
I don't know of any, and I like this update! :crossed_fingers: :grimacing: 

I would maybe rephrase the message to be more explicit about how to fix this:

The use of `map_overlap(array, function, **kwargs)` is deprecated since dask x.y and will be an error in dask z.w. To silence this message, use the syntax `map_overlap(function, array0,[ array1, ...,] **kwargs)`.

Reply:
Thanks @jni, I'll make that change.

Reply:
I updated the message assuming that this change would be pushed in 2.17.  Not sure if that will be true, and I also wasn't sure when to say it would be an error so I hedged by saying it would happen in some future release.  This message seem reasonable now?